src/OmnichannelChatSDK.ts

```diff
@@ -1429,12 +1429,25 @@ class OmnichannelChatSDK {
         let match;
 
         for (const maskingRule of this.dataMaskingRules.rules) {
-            const regex = new RegExp(maskingRule.regex, 'g');
-            while ((match = regex.exec(content)) !== null) {
-                const replaceStr = match[0].replace(/./g, this.maskingCharacter);
-                content = content.replace(match[0], replaceStr);
+            try {
```

Comment:
Explain what would happen here This isn't working:
(?<!\d)(?<!+)(?<!+\d)(?<!+\d{2})(?<!+\d{3})(?<!+\d\s)(?<!+\d{2}\s)(?<!+\d{3}\s)(?:[1-9]\d{2}[-\s]?\d{3}[-\s]?\d{3}|[1-9]\d{8})(?!\d)
 
However, after modifying it to the following, it worked:
\b[1-9]\d{2}[-\s]?\d{3}[-\s]?\d{3}\b